src/Microsoft.ML.AutoML/API/ExperimentResults/CrossValidationExperimentResult.cs

```diff
@@ -35,6 +36,46 @@ public class CrossValidationExperimentResult<TMetrics>
         {
             RunDetails = runDetails;
             BestRun = bestRun;
+            _disposed = false;
+            _disposedRunDetails = false;
         }
+
+        #region IDisposable Support
+        private bool _disposed;
+        private bool _disposedRunDetails;
+
+        /// <summary>
+        /// Releases unmanaged Tensor objects in models stored in RunDetail and BestRun instances
+        /// </summary>
+        /// <remarks>
+        /// Invocation of Dispose() is necessary to clean up remaining C library Tensor objects and
+        /// avoid a memory leak
+        /// </remarks>
+        public void Dispose()
+        {
+            if (_disposed)
+                return;
+            if (!_disposedRunDetails)
+                (RunDetails as IDisposable)?.Dispose();
+            (BestRun as IDisposable)?.Dispose();
+            _disposed = true;
+            _disposedRunDetails = true;
+        }
+
+        /// <summary>
+        /// Releases unmanaged Tensor objects in models stored in RunDetail instances
+        /// </summary>
```

Comment:
Please add a more detailed description here that this is provided to support use cases where the user might want different lifetimes for the best model and the rest of the experimental runs.